wallet/wallet-cli/src/repl/mod.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::{Command, FromArgMatches, Subcommand};
+use node_comm::node_traits::NodeInterface;
+use reedline::{
+    ColumnarMenu, DefaultCompleter, DefaultHinter, DefaultValidator, EditMode, Emacs,
+    ExampleHighlighter, FileBackedHistory, ListMenu, Reedline, ReedlineMenu, Signal, Vi,
+};
+
+use crate::{
+    cli_println,
+    commands::{handle_wallet_command, WalletCommands},
+    config::WalletCliConfig,
+    errors::WalletCliError,
+    output::OutputContext,
+    repl::wallet_prompt::WalletPrompt,
+    DefWallet,
+};
+
+mod wallet_prompt;
+
+const HISTORY_FILE_NAME: &str = "history.txt";
+const HISTORY_MAX_LINES: usize = 1000;
+
+fn get_repl_command() -> Command {
+    // Strip out usage
+    const PARSER_TEMPLATE: &str = "\
+        {all-args}
+    ";
+
+    // Strip out name/version
+    const APPLET_TEMPLATE: &str = "\
+        {about-with-newline}\n\
+        {usage-heading}\n    {usage}\n\
+        \n\
+        {all-args}{after-help}\
+    ";
+
```

Comment:
How difficult is it to support `help <cmd>`, where we can add additional information about the arguments, and such?

Reply:
It already works.

Reply:
Can we add extra information to the returned message that only appears when we `help <cmd>`?

Reply:
Good question. I just tried and it can be done like this:
```
    /// Submit a block to be included in the chain
    ///
    /// More information about block submits.
    /// More information about block submits.
    ///
    /// Even more information about block submits.
    /// Even more information about block submits.
    /// Even more information about block submits.
    /// Even more information about block submits.
    SubmitBlock {
        /// Hex encoded block
        block: String,
    },
```

```
Wallet> help submitblock
Submit a block to be included in the chain

More information about block submits. More information about block submits.

Even more information about block submits. Even more information about block submits. Even more information about block submits. Even more information about block submits.

Usage:
    submitblock <BLOCK>

Arguments:
  <BLOCK>
          Hex encoded block

Options:
  -h, --help
          Print help (see a summary with '-h')
```

And the short help works like before:
```
Wallet> help 
Commands:
  bestblock        Returns the current best block hash
  bestblockheight  Returns the current best block height
  submitblock      Submit a block to be included in the chain
  rescan           Rescan
  exit             Quit the REPL
  history          Print history
  clear            Clear screen
  clearhistory     Clear history
  help             Print this message or the help of the given subcommand(s)
```

Reply:
Very nice! Please add something as an example and we'll complete the docs later.